pkg/providers/cloudstack/template_test.go

```diff
@@ -124,3 +139,14 @@ func TestTemplateBuilderGenerateCAPISpecWorkersInvalidSSHKey(t *testing.T) {
 		MatchError(ContainSubstring("formatting ssh key for cloudstack worker template: ssh")),
 	)
 }
+
+func TestTemplateBuilderGenerateCAPISpecWorkersInvalidEndpoint(t *testing.T) {
+	g := NewWithT(t)
+	templateBuilder := cloudstack.NewTemplateBuilder(time.Now)
+	clusterSpec := test.NewFullClusterSpec(t, path.Join(testDataDir, testClusterConfigMainFilename))
+	clusterSpec.Cluster.Spec.ProxyConfiguration = &v1alpha1.ProxyConfiguration{}
+	clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host = "1.1.1.1::"
+	machineTemplateNames, kubeadmConfigTemplateNames := clusterapi.InitialTemplateNamesForWorkers(clusterSpec)
+	_, err := templateBuilder.GenerateCAPISpecWorkers(clusterSpec, machineTemplateNames, kubeadmConfigTemplateNames)
+	g.Expect(err).To(MatchError(ContainSubstring("building template map for MD host 1.1.1.1:: is invalid: address 1.1.1.1::: too many colons in address")))
+}
```

Comment:
Can we have two more success test cases to check the default port is not provided but specified in in the template or capi object

Reply:
Those test cases are currently being covered by the success tests in `cloudstack_test.go`. Do you think it's still worth it to add coverage for them here?